### PR TITLE
python3Packages.remi: init at 2022.7.27

### PR DESCRIPTION
**Pull request by pbsds**
###### Description of changes

[REMI](https://github.com/rawpython/remi) is a simple library to build dynamic websocket-driven webuis.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/remi/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    hash = "sha256-VC34jhdLvVj+djxJ5cIRbyyEE8GpNHMQae2MF7uc1kw=";
+  };
+
+  nativeBuildInputs = [
+    setuptools-scm
+  ];
+
+  # We're unable to deselect "optional" integrations tests
+  # with unpackaged libraries (pyepics)
+  doCheck = false;
```
Comment (risicle) [inline pkgs/development/python-modules/remi/default.nix:23]: With `unittest` tests, it's possible to "disable" individual tests by using e.g. `substituteInPlace` to make them either not start with `test_` or `Test` (I tend to turn them into e.g. `dont_test_`), and `unittest` won't pick them up.

Comment (SuperSandro2000) [reply pkgs/development/python-modules/remi/default.nix:23]: How hard would it be to package pyepics?

Comment (pbsds) [reply pkgs/development/python-modules/remi/default.nix:23]: Packaging pyepics is far from easy. There are many forks and dead dependencies. For a simple widget test it is not worth it.
I managed to deselect the test using pytest and lots of patching.

**Comment by pbsds** [comment]
Changes with this force push:

* fetches sources from github instead of pypi
* removed depencency on setuptools-scm
* enabled tests
* patch lots of broken imports

**Comment by SuperSandro2000** [comment]
@ofborg build python310Packages.remi

**Pull request merged**